### ISSUE TITLE
Passing a DOM element to `decode()` causes infinite recursion

### ISSUE THREAD

**Issue by passcod** [ParsePlatform/Parse-SDK-JS#181]
TL;DR: `decode()` (in my case invoked through `Parse.User.logIn()` is obviously not meant be used with a DOM element, but it would be useful to have it clearly fail/throw/return null instead of causing a recursion error which is very confusing.

---

This is using 1.6.14 from npm, passing a dom element like:

```js
Parse.User.logIn(e.target.username, e.target.password)...
```

Obviously the "right way" is:

```js
Parse.User.logIn(e.target.username.value, e.target.password.value)...
```

but the error provided (`too much recursion` in Firefox) is less than helpful and requires reading through the Parse source and understanding what it does to actually "get it".

This issue is _not_ asking for `decode()` to magically handle DOM elements, it's asking for a better error message if something is passed to `decode()` that is not supported.

**Comment by andrewimm** [comment]
Ideally, nothing is unsupported by decode. If any item passed in is a viable JS object, it is expected to iterate over the keypairs. We could change the behavior to check if the constructor of an object is `Object` (true for pure objects, false for other browser builtins like Node or Element), but I'd be worried about breaking some behavior with such a heavy-handed approach.
It seems like what you really want in this case is validation to make sure that the values passed to `Parse.User.logIn`, which is something I agree would improve the developer experience. I'd be happy to add validation to make sure the inputs are strings.

**Comment by passcod** [comment]
Yeah, that would work :)

**Comment by andrewimm** [comment]
Landed in 1.7.1
